test/integration/custom.js

```diff
@@ -0,0 +1,223 @@
+var indico = require('../..')
+  , settings = require('../../lib/settings.js')
+  , should = require('chai').should()
+  , expect = require('chai').expect
+  ;
+
+describe('Custom', function() {
+  this.timeout(35000);
+
+  var testData = [
+    ['aidan rules', 'test_writer'],
+    ['sometimes tests are fun', 'test_writer'],
+    ['meow meow meow', 'aidan']
+  ];
+
+  if (settings.resolveApiKey() === false) {
+    // skip test -- indico auth keys are not available
+    console.warn('Api keys are now required. Skipping some tests.\nhttp://docs.indico.io/v2.0/docs/api-keys')
+    return;
+  }
+  
+  describe('addData', function() {
+    it('should add a single example to a collection', function(done) {
+      indico.Collection("test").clear().then(function() {
+        var testCollection = indico.Collection("test");
+        
+        testCollection.info().then(function(res) {
+          res['status'].should.equal('no examples');
```

Comment:
Isn't this promises written in the style of callbacks?  Couldn't we close the first then before we introduce a second.